### ISSUE TITLE
Channel Rebalancing 

### ISSUE THREAD

**Issue by chriso22** [ZeusLN/zeus#2374]
### Describe the bug

A bug happened! I attempted to rebalance channels by creating a receive invoice, then paying that invoice and selecting the 1st hop (move outbound from) and last hop (move outbound to) and the invoice payment doesn’t follow the instructions. The invoice route out and back in to the same channel, which happens to be listed directly below the channel I was attempting to send outbound liquidity to. 

This happened 3 times in consecutively.  I updated to the version 0.9.0 prior to this rebalance. This is the first time I’ve ever experienced this issue. 

### Reproduce

I attempted to rebalance channels by creating a receive invoice, then paying that invoice and selecting the 1st hop (move outbound from) and last hop (move outbound to) and the invoice payment doesn’t follow the instructions. The invoice route out and back in to the same channel, which happens to be listed directly below the channel I was attempting to send outbound liquidity to. 

### ZEUS version

v0.9.0

### Node interface

LND (Lightning Node Connect)

### Network

Tor

### Device

iPhone 14

### Device operating system

17.5.1

**Comment by kaloudis** [comment]
Are you creating wrapped invoices with the LSP enabled?

**Comment by kaloudis** [comment]
It would be interesting to see if you downgraded to v0.8.5-hotfix and observed the behavior

**Comment by chriso22** [comment]
Evan,If I’m creating wrapped invoices, I didn’t know that I am.  I’m not doing anything different than I’ve done in the past.  Sent from my iPhoneOn Aug 27, 2024, at 8:06 PM, Evan Kaloudis ***@***.***> wrote:﻿
Are you creating wrapped invoices with the LSP enabled?

**Comment by chriso22** [comment]
I’ll try that tomorrow. I thought about that as well. Will follow up once I do that. Sent from my iPhoneOn Aug 27, 2024, at 8:06 PM, Evan Kaloudis ***@***.***> wrote:﻿
It would be interesting to see if you downgraded to v0.8.5-hotfix and observed the behavior

**Comment by chriso22** [comment]
I’ve looked around and it doesn’t appear that I can downgrade my version after updating. I’ve honestly never done it before so maybe I’m missing something.Sent from my iPhoneOn Aug 27, 2024, at 8:06 PM, Evan Kaloudis ***@***.***> wrote:﻿
It would be interesting to see if you downgraded to v0.8.5-hotfix and observed the behavior

**Comment by kaloudis** [comment]
@chriso22 having problems recreating.

are you on Umbrel? Could you provide your LND and Lightning Terminal version numbers?

**Comment by chriso22** [comment]
Evan,I’m on Start9. I’m using LND 0.18.0.1 and LiT 0.13.0. Sent from my iPhoneOn Aug 29, 2024, at 8:13 PM, Evan Kaloudis ***@***.***> wrote:﻿
@chriso22 having problems recreating.
are you on Umbrel? Could you provide your LND and Lightning Terminal version numbers?

**Comment by kaloudis** [comment]
> Evan,I’m on Start9. I’m using LND 0.18.0.1 and LiT 0.13.0. Sent from my iPhoneOn Aug 29, 2024, at 8:13 PM, Evan Kaloudis ***@***.***> wrote:﻿ @chriso22 having problems recreating. are you on Umbrel? Could you provide your LND and Lightning Terminal version numbers? —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

Thanks for reporting? Any chance newer versions are available on StartOS?

Latest are LND v0.18.2-beta and LITD v0.13.3-beta

**Comment by chriso22** [comment]
I just had a notice today that I have an update for LiT. But there’s nothing for LND.  I haven’t updated it yet. I like to give it a little time before updating. Sent from my iPhoneOn Aug 30, 2024, at 8:33 PM, Evan Kaloudis ***@***.***> wrote:﻿

Evan,I’m on Start9. I’m using LND 0.18.0.1 and LiT 0.13.0. Sent from my iPhoneOn Aug 29, 2024, at 8:13 PM, Evan Kaloudis @.> wrote:﻿ @chriso22 having problems recreating. are you on Umbrel? Could you provide your LND and Lightning Terminal version numbers? —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: @.>

Thanks for reporting? Any chance newer versions are available on StartOS?
Latest are LND v0.18.2-beta and LITD v0.13.3-beta

**Comment by ElucGeek** [comment]
Hello, I have the same issue, I select the first hop and last hop to do rebalancing and it's completely ignoring it and as above, it's just going in and out of the same channel, resulting in a fee for nothing changed. What is even more surprising is that the peer of the first and last hop selected have multiple common channels, of course I don't know their liquidity but there should be a path using one or multiple channel between them directly.
Is there a way to force the first hop last hop and only route with these or fail if not possible?

**Comment by kaloudis** [comment]
Yes, that’s the intended functionality. Can you provide your LND version and which interface you’re using? If you’re using LNC can you provide your Lightning Terminal version?

**Comment by kaloudis** [comment]
@ElucGeek Yes, that’s the intended functionality. Can you provide your LND version and which interface you’re using? If you’re using LNC can you provide your Lightning Terminal version?

**Comment by kaloudis** [comment]
We've figured it out @ElucGeek @chriso22 - please test out v0.9.1-rc2 if you have the time

**Comment by ElucGeek** [comment]
OK it works, thank you for the fix!

**Comment by chriso22** [comment]
Hey Evan, glad to hear it’s fixed. I’m not getting the option to update yet. But I’ll be sure and test it once I do.   Sent from my iPhoneOn Oct 10, 2024, at 11:59 AM, Evan Kaloudis ***@***.***> wrote:﻿
We've figured it out @ElucGeek @chriso22 - please test out v0.9.1-rc2 if you have the time

**Comment by kaloudis** [comment]
v0.9.1 is now available with the fix: https://github.com/ZeusLN/zeus/releases/tag/v0.9.1

**Comment by chriso22** [comment]
Evan,

 

Updated and can confirm it is back to working like it should. Thanks for keeping me updated on the progress.

 

Chris
